llama-cpp-2/src/sampling.rs

```diff
@@ -20,6 +20,15 @@ impl Debug for LlamaSampler {
     }
 }
 
+// this is needed for the dry sampler to typecheck on android
+// ...because what is normally an i8, is an u8 
+#[cfg(target_os = "android")]
+type CChar = u8;
+
+#[cfg(not(target_os = "android"))]
+type CChar = i8;
+
+
```

Comment:
```suggestion
// this is needed for the dry sampler to typecheck on android
// ...because what is normally an i8, is an u8 
#[cfg(target_os = "android")]
type CChar = u8;

#[cfg(not(target_os = "android"))]
type CChar = i8;


```